src/report.rs

```diff
@@ -190,6 +190,78 @@ impl Report for TapReport {
     fn done(&mut self) {}
 }
 
+/// This is a particularly simpl reporter, mostly useful with --nocapture,
+/// that will collect stats and results and will display a summary
+/// after all tests are done.
+pub struct CompactReport {
+    writer: ColorWriter,
+    results: BTreeMap<String, (Duration, Status)>,
+    stats: TestStats,
+}
+
+impl CompactReport {
+    pub fn new(writer: ColorWriter) -> Self {
+        Self {
+            writer,
+            results: BTreeMap::new(),
+            stats: Default::default(),
+        }
+    }
+}
+
+impl Report for CompactReport {
+    fn init(&mut self, _plan: &[Task]) {}
+
+    fn start(&mut self, _task_name: String) {}
+
+    fn report(&mut self, result: &CompletedTask) {
+        self.stats.update(result);
+        self.results.insert(
+            result.full_name.join("::"),
+            (result.duration, result.status.clone()),
+        );
+    }
+
+    fn done(&mut self) {
+        for (name, (dur, res)) in self.results.iter() {
+            let (lbl, color) = match res {
+                Status::Success => ("pass", GREEN),
+                Status::Failure(_) => ("fail", RED),
+                Status::Signaled(_) => ("sign", YELLOW),
+                Status::Timeout => ("tout", RED),
```

Comment:
> tout

I'd never have guessed what that means :)
Consider using longer labels, padding them with spaces if needed.

Reply:
sure